saleor/payment/gateways/np_atobarai/api.py

```diff
@@ -0,0 +1,209 @@
+    """Resolve test/production URLs based on the api config."""
+    return f"{(NP_TEST_URL if config.test_mode else NP_URL)}{path}"
+
+
+def np_request(
+    config: ApiConfig, method: str, path: str = "", json: Optional[dict] = None
+) -> requests.Response:
+    try:
+        return requests.request(
+            method=method,
+            url=get_url(config, path),
+            timeout=REQUEST_TIMEOUT,
+            json=json or {},
+            auth=HTTPBasicAuth(config.merchant_code, config.sp_code),
+            headers={"X-NP-Terminal-Id": config.terminal_id},
+        )
+    except requests.RequestException:
+        raise PaymentError("Cannot connect to NP Atobarai.")
+
+
+def health_check(config: ApiConfig) -> bool:
+    try:
+        response = np_request(config, "post", "/authorizations/find")
+        return response.status_code not in [401, 403]
+    except PaymentError:
+        return False
+
+
+def _format_name(ad: AddressData):
+    """Follow the japanese name guidelines."""
+    return f"{ad.first_name} {ad.last_name}".lstrip().rstrip()
+
+
+def _format_address(ad: AddressData):
+    """Follow the japanese address guidelines."""
+    # example: "東京都千代田区麹町４－２－６　住友不動産麹町ファーストビル５階"
+    with Posuto() as pp:
+        try:
+            jap_ad = pp.get(ad.postal_code)
+        except KeyError:
+            raise PaymentError(
+                "Valid japanese address is required for transaction in NP Atobarai."
+            )
+        else:
+            return (
+                f"{ad.country_area}"
+                f"{jap_ad.city}"
+                f"{jap_ad.neighborhood}"
+                f"{ad.street_address_2}"
+                f"{ad.street_address_1}"
+            )
+
+
+def register_transaction(
+    config: ApiConfig, payment_information: "PaymentData"
+) -> PaymentResult:
+    order_date = timezone.now().strftime("%Y-%m-%d")
+
+    billing = payment_information.billing
+    shipping = payment_information.shipping
+
+    if not billing:
+        raise PaymentError(
+            "Billing address is required for transaction in NP Atobarai."
+        )
+    if not shipping:
+        raise PaymentError(
+            "Shipping address is required for transaction in NP Atobarai."
+        )
+
+    data = {
+        "transactions": [
+            {
+                "shop_transaction_id": payment_information.payment_id,
+                "shop_order_date": order_date,
+                "settlement_type": NP_ATOBARAI,
+                "billed_amount": int(
+                    price_to_minor_unit(
+                        payment_information.amount, payment_information.currency
+                    )
+                ),
+                "customer": {
+                    "customer_name": billing.first_name,
+                    "company_name": billing.company_name,
+                    "zip_code": billing.postal_code,
+                    "address": _format_address(billing),
+                    "tel": billing.phone.replace("+81", "0"),
```

Comment:
Is phone mandatory? I think we should have an extra validation higher up that ensures that:
1. The phone number is provided
2. It is a Japanese phone number

Reply:
checkoutCreate is already validating the phone number